### ISSUE TITLE
The Image plug-in displays abnormal images in certain sizes

### ISSUE THREAD

**Issue by qisongH** [ros-visualization/rviz#1762]
When dragging the Image plug-in, **a row or column of abnormal pixels sometimes appear at the boundary of the image**. The following figure shows the exception.

![image](https://user-images.githubusercontent.com/74614539/182803257-36a41fa3-2d4b-4343-aad9-ed106528f89c.png)
The size of the Image plug-in is about *240×217*, and the **lower boundary pixels** are displayed incorrectly.

![image](https://user-images.githubusercontent.com/74614539/182803279-7a5394de-55cd-4e69-9b7a-ae6188b59f0b.png)
The size of the Image plug-in is about *433×155*, and the **right boundary pixels** are displayed incorrectly.

### Your environment
* OS Version: e.g. Ubuntu 18.04
* ROS Distro: [Melodic]
* RViz, Qt, OGRE, OpenGl version as printed by rviz:
```
 [ INFO] [1646653665.369882000]: rviz version 1.13.18
 [ INFO] [1646653665.369933300]: compiled against Qt version 5.9.5
 [ INFO] [1646653665.369944800]: compiled against OGRE version 1.9.0 (Ghadamon)
 [ INFO] [1646653665.373622300]: Forcing OpenGl version 0.
 [ INFO] [1646653665.491788300]: Stereo is NOT SUPPORTED
 [ INFO] [1646653665.492296300]: OpenGL device: llvmpipe (LLVM 10.0.0, 256 bits)
 [ INFO] [1646653665.492382900]: OpenGl version: 3.1 (GLSL 1.4).
```

* OS Version: e.g. Ubuntu 20.04
* ROS Distro: [Noetic]
* RViz, Qt, OGRE, OpenGl version as printed by rviz:
```
[ INFO] [1659601560.538959700]: rviz version 1.14.14
[ INFO] [1659601560.539014300]: compiled against Qt version 5.12.8
[ INFO] [1659601560.539023500]: compiled against OGRE version 1.9.0 (Ghadamon)
[ INFO] [1659601560.547179000]: Forcing OpenGl version 0.
[ INFO] [1659601560.706482400]: Stereo is NOT SUPPORTED
[ INFO] [1659601560.706561400]: OpenGL device: llvmpipe (LLVM 12.0.0, 256 bits)
[ INFO] [1659601560.706581800]: OpenGl version: 3.1 (GLSL 1.4).
```

**Comment by rhaschke** [comment]
Thanks for reporting. Could you please review #1763?

**Comment by qisongH** [comment]
> Thanks for reporting. Could you please review #1763?

Thanks for the fix.

**Comment by rhaschke** [comment]
Closed via #1763.
